### ISSUE TITLE
Issues with URL when wallet does not support several networks

### ISSUE THREAD

**Issue by elena-zh** [cowprotocol/cowswap#1391]
Related to #1317 and #1384

Reported in https://github.com/cowprotocol/cowswap/pull/1384#issuecomment-1314990553 (cases 3-4)

3. Tokens in the link are not removed when I connect a wallet using WC, and it does not support other chains but Ethereum (example: TrustWallet): 
![return back](https://user-images.githubusercontent.com/70885163/201908699-bbf4021e-96b3-4a68-9eba-ea2c64f09223.png)

4. When I change a network indicator in the URL (to 4, 5, 100) being connected to a wallet that does not support a network switch (supports only 1), it would be great to return the indicator to 1.

**Comment by anxolin** [comment]
In general, the experience with Trust wallet was not too bad in the test i just did. 

3. For me in BARN its removing the tokens, and selecting the default in the network. Not sure I got the issue right, if I did, it think is ok

4. I agree, this is not too good. I also see a problem that the networkId of the wallet don't match the one in the UI. Or if we allow this, we should at least show a warning and tell the user to change the network in their wallet or something.

Anyways, leaving this open, not blocking the release. We should revisit this though!
